incubator/binding-mqtt-kafka/src/main/java/io/aklivity/zilla/runtime/binding/mqtt/kafka/internal/stream/MqttKafkaSubscribeFactory.java

```diff
@@ -194,7 +217,18 @@ private final class MqttSubscribeProxy
         private long replyAck;
         private int replyMax;
         private int replyPad;
-
+        private long budgetId;
+        private int mqttSharedBudget;
+
+        private final Queue<List<Subscription>> deferredRetainedFiltersQueue = new LinkedList<>();
+        private final List<Subscription> retainedFilters = new ArrayList<>();
+        private final IntArrayList messagesSubscriptionIds = new IntArrayList();
+        private final IntArrayList retainedSubscriptionIds = new IntArrayList();
+        private final Long2ObjectHashMap<Boolean> retainAsPublished = new Long2ObjectHashMap<>();
```

Comment:
Fields that are not non-final constants should be initialized in the constructor please.